packages/connector-alipay/src/index.ts

```diff
@@ -0,0 +1,227 @@
+      app_id,
+      redirect_uri, // The variable `redirectUri` should match {appId, appSecret}
+      scope,
+      state,
+    })}`;
+  };
+
+  public getAccessToken: GetAccessToken = async (code): Promise<AccessTokenObject> => {
+    const config = await this.getConfig(this.metadata.id);
+    const initSearchParameters = {
+      method: methodForAccessToken,
+      format: 'JSON',
+      timestamp: this.getTimestamp(),
+      version: '1.0',
+      grant_type: 'authorization_code',
+      code,
+      charset: 'UTF8',
+      ...config,
+    };
+    const signedSearchParameters = this.signingPamameters(initSearchParameters);
+
+    const response = await got
+      .post(alipayEndpoint, {
+        searchParams: signedSearchParameters,
+        timeout: await this.getRequestTimeout(),
+      })
+      .json<AccessTokenResponse>();
+
+    const { msg, sub_msg } = response.error_response ?? {};
+    assert(
+      response.alipay_system_oauth_token_response,
+      new ConnectorError(ConnectorErrorCodes.SocialAuthCodeInvalid, msg ?? sub_msg)
+    );
+    const { access_token: accessToken } = response.alipay_system_oauth_token_response;
+
+    assert(accessToken, new ConnectorError(ConnectorErrorCodes.SocialAuthCodeInvalid));
+
+    return { accessToken };
+  };
+
+  public getUserInfo: GetUserInfo = async (accessTokenObject) => {
+    const config = await this.getConfig(this.metadata.id);
+    const { accessToken } = accessTokenObject;
+    assert(
+      accessToken && config,
+      new ConnectorError(ConnectorErrorCodes.InsufficientRequestParameters)
+    );
+
+    const initSearchParameters = {
+      method: methodForUserInfo,
+      format: 'JSON',
+      timestamp: this.getTimestamp(),
+      version: '1.0',
+      grant_type: 'authorization_code',
+      auth_token: accessToken,
+      biz_content: JSON.stringify({}),
+      charset: 'UTF8',
+      ...config,
+    };
+    const signedSearchParameters = this.signingPamameters(initSearchParameters);
+
+    const response = await got
+      .post(alipayEndpoint, {
+        searchParams: signedSearchParameters,
+        timeout: await this.getRequestTimeout(),
+      })
+      .json<UserInfoResponse>();
+
+    const {
+      user_id: id,
+      avatar,
+      nick_name: name,
+      sub_msg,
+      sub_code,
+      msg,
+      code,
+    } = response.alipay_user_info_share_response;
+
+    if (sub_msg || sub_code) {
+      if (code === '20001') {
+        throw new ConnectorError(ConnectorErrorCodes.SocialAccessTokenInvalid, msg);
+      }
+      throw new ConnectorError(ConnectorErrorCodes.General);
+    }
+    // TODO: elaborate on the error messages for all social connectors (see LOG-2157)
+
+    assert(id, new ConnectorError(ConnectorErrorCodes.InvalidResponse));
+
+    return { id, avatar, name };
+  };
+
+  // Reference: https://github.com/alipay/alipay-sdk-nodejs-all/blob/10d78e0adc7f310d5b07567ce7e4c13a3f6c768f/lib/util.ts
+  public readonly signingPamameters = (
```

Comment:
can be private?

Reply:
Private methods can not be mocked. This could bring trouble when writing UTs, thought for a while but can not find a good way to address this.
[LOG-2283](https://linear.app/silverhand/issue/LOG-2283/check-connector-classes-method-modifier) was added to track this.